path_oidc.go

```diff
@@ -243,7 +243,15 @@ func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request,
 	}
 
 	if role.VerboseOIDCLogging {
-		b.Logger().Debug("OIDC provider response", "ID token", rawToken)
+		loggedToken := "invalid token format"
+
+		parts := strings.Split(rawToken, ".")
+		if len(parts) == 3 {
+			// strip signature from logged token
+			loggedToken = fmt.Sprintf("%s.%s.xxxxxxxxxxx", parts[0], parts[1])
```

Comment:
I wish this could be represented as a valid JWT token so that it could be parsed, but this probably the cheap way to do it without having to construct the signature from scratch using an empty secret value.

Reply:
The popular parsers still show the claims info and structure, just with a warning about the signature. And you can always just grab a segment (probably the middle one) and pass it in to a base64 decoder directly.

Reply:
Retracting my initial comments after giving it some more thought. Making it look more like the actual JWT might end up being worse UX since the user can't tell if that's the actual JWT token or not (they might assume it is but it's not) without digging into the specifics of how this value is constructed.